### PR TITLE
Single game page/feature/#5

### LINKED ISSUE #12: Make game page

### PR DESCRIPTION
**Pull request by drmodun**
closes #12 
The first draft for single game page, should have enough info now, of course reviews to be added later

### CODE REVIEW

File: pub-critic/src/app/api/GamesShared.ts [hunk 1]
```diff
@@ -4,6 +4,21 @@ export const apiKey = "464bc085dbbf4f33bcb2ccb39d36a6ec";
...
+  released: string;
+  background_image: string;
+  description: string;
+  rating: number;
+  platforms: PlatformFull[];
+  website: string;
+  metacritic_url: string;
 }
 
 export interface Genre {
```
Comment (ilovric00) [inline pub-critic/src/app/api/GamesShared.ts]: `app/interfaces`

File: pub-critic/src/app/components/GameInfo/GameInfo.tsx [hunk 2]
```diff
@@ -0,0 +1,89 @@
+"use client";
+
+import { Screenshots } from "components/Screenshots/Screenshots";
+import classes from "./GameInfo.module.scss";
+import { Platform, PlatformFull, Screenshot } from "api/GamesShared";
+import Image from "next/image";
+import contoller from "assets/controller.svg";
+interface GameInfoProps {
```
Comment (ilovric00) [inline pub-critic/src/app/components/GameInfo/GameInfo.tsx:10]: missing space before

File: pub-critic/src/app/components/GameInfo/GameInfo.tsx [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+  metacritic,
+  rating,
+  background_image,
+  platforms,
+  released,
+  website,
+  metacritic_url,
+  screenshots,
+}: GameInfoProps) => {
+  console.log(screenshots);
```
Comment (ilovric00) [inline pub-critic/src/app/components/GameInfo/GameInfo.tsx]: Remove unnecessary console log

File: pub-critic/src/app/components/GameInfo/GameInfo.tsx [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+              <div className={classes.platform} key={platform.platform.id}>
+                {platform.platform.name}
+              </div>
+            ))}
+          </div>
+          <div className={classes.released}>
+            <span>Released: {released} </span>
+          </div>
+          <div className={classes.buttons}>
+            <a href={website} className={classes.link}>
```
Comment (ilovric00) [inline pub-critic/src/app/components/GameInfo/GameInfo.tsx]: use Next Link component

File: pub-critic/src/app/components/Screenshots/Screenshots.tsx [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+          src={screenshots[selected] ? screenshots[selected].image : contoller}
+          alt="screenshot"
+          layout="fill"
+        />
+      </div>
+      <div className={classes.screenshots}>
+        {screenshots.map((screenshot, index) => (
+          <div className={classes.screenshot} key={screenshot.id}>
+            <Image
+              className={index === selected ? classes.selected : ""}
```
Comment (ilovric00) [inline pub-critic/src/app/components/Screenshots/Screenshots.tsx]: `clsx` lib can help here

File: pub-critic/src/app/components/Screenshots/Screenshots.tsx [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+              key={index}
+              src={screenshot.image}
+              alt="screenshot"
+              layout="fill"
+              objectFit="cover"
+              onClick={() => setSelected(index)}
```
Comment (ilovric00) [inline pub-critic/src/app/components/Screenshots/Screenshots.tsx]: `handleSelect` method

File: pub-critic/src/app/games/[gameId]/page.tsx [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+  if (response) {
+    return response;
+  }
+};
+
+const fetchGameScreenshots = async (gameId: number) => {
+  const response = await getScreenshots(gameId);
+  if (response) {
+    return response;
+  }
```
Comment (ilovric00) [inline pub-critic/src/app/games/[gameId]/page.tsx]: if no response? :)

File: pub-critic/src/app/games/[gameId]/page.tsx [hunk 8]
```diff
@@ -0,0 +1,45 @@
...
+};
+
+const GamePage = async ({ params }: { params: any }) => {
+  const game = await fetchGameDetails(params.gameId);
+  const screenshots = await fetchGameScreenshots(params.gameId);
+  console.log(screenshots);
```
Comment (ilovric00) [inline pub-critic/src/app/games/[gameId]/page.tsx]: remove console logs you don't need

**Review by ilovric00 — CHANGES_REQUESTED** [review]
few things here as well...GGWP 💪

File: pub-critic/src/app/api/GamesShared.ts (hunk 1 above)
Comment (drmodun) [reply pub-critic/src/app/api/GamesShared.ts]: ✅

File: pub-critic/src/app/components/GameInfo/GameInfo.tsx (hunk 2 above)
Comment (drmodun) [reply pub-critic/src/app/components/GameInfo/GameInfo.tsx:10]: ✅

File: pub-critic/src/app/components/GameInfo/GameInfo.tsx (hunk 4 above)
Comment (drmodun) [reply pub-critic/src/app/components/GameInfo/GameInfo.tsx]: ✅

File: pub-critic/src/app/games/[gameId]/page.tsx (hunk 7 above)
Comment (drmodun) [reply pub-critic/src/app/games/[gameId]/page.tsx]: ✅

File: pub-critic/src/app/games/[gameId]/page.tsx (hunk 8 above)
Comment (drmodun) [reply pub-critic/src/app/games/[gameId]/page.tsx]: ✅

File: pub-critic/src/app/components/Screenshots/Screenshots.tsx (hunk 6 above)
Comment (drmodun) [reply pub-critic/src/app/components/Screenshots/Screenshots.tsx]: ✅

File: pub-critic/src/app/components/Screenshots/Screenshots.tsx (hunk 5 above)
Comment (drmodun) [reply pub-critic/src/app/components/Screenshots/Screenshots.tsx]: ✅

File: pub-critic/src/app/components/GameInfo/GameInfo.tsx (hunk 3 above)
Comment (drmodun) [reply pub-critic/src/app/components/GameInfo/GameInfo.tsx]: ✅

File: pub-critic/src/app/api/GamesApi.ts [hunk 9]
```diff
@@ -1,4 +1,5 @@
-import { apiKey, gamesApi } from "./GamesShared";
+export const gamesApi = "https://api.rawg.io/api";
+export const apiKey = "464bc085dbbf4f33bcb2ccb39d36a6ec";
```
Comment (kula124) [inline pub-critic/src/app/api/GamesApi.ts:2]: I have your key now :)

File: pub-critic/src/app/components/GameInfo/GameInfo.tsx [hunk 10]
```diff
@@ -0,0 +1,90 @@
...
+  metacritic,
+  rating,
+  background_image,
+  platforms,
+  released,
+  website,
+  metacritic_url,
+  screenshots,
+}: GameInfoProps) => {
+  return (
```
Comment (kula124) [inline pub-critic/src/app/components/GameInfo/GameInfo.tsx]: You can omit return here

File: pub-critic/src/app/components/Screenshots/index.ts
Comment (kula124) [inline pub-critic/src/app/components/Screenshots/index.ts:1]: ?

File: pub-critic/src/app/components/GameInfo/GameInfo.module.scss [hunk 11]
```diff
@@ -0,0 +1,256 @@
+@import "styles/variables.scss";
+@import "styles/vw-calc.scss";
```
Comment (ilovric00) [inline pub-critic/src/app/components/GameInfo/GameInfo.module.scss:2]: you can add these frequent files to `next.config.js` file so you don't have to import each time:

```
/// next.config.js
const nextConfig = {
  sassOptions: {
    prependData: `
        @import 'styles/utils/rem-calc';
        @import 'styles/settings/variables';
        @import 'styles/mixins/_breakpoints.scss';
        ${process.env.NEXT_PUBLIC_BASEPATH ? `$basepath: ${process.env.NEXT_PUBLIC_BASEPATH}` : "$basepath: ''"};
    `,
  },
  basePath: process.env.NEXT_PUBLIC_BASEPATH,
};

if (process.env.ANALYZE === 'true') {
  const bundleAnalyerLocal = '@next/bundle-analyzer';
  const withBundleAnalyzer = require(bundleAnalyerLocal)({
    enabled: true,
  });

  module.exports = withBundleAnalyzer(nextConfig);
} else {
  module.exports = nextConfig;
}
```

**Review by ilovric00 — APPROVED** [review]

File: pub-critic/src/app/components/GameInfo/GameInfo.module.scss (hunk 11 above)
Comment (drmodun) [reply pub-critic/src/app/components/GameInfo/GameInfo.module.scss:2]: gonna remember this for future use cases, going in my refactory list

File: pub-critic/src/app/components/GameInfo/GameInfo.tsx (hunk 10 above)
Comment (drmodun) [reply pub-critic/src/app/components/GameInfo/GameInfo.tsx]: :white_check_mark:

**Pull request merged**